apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -27,215 +24,210 @@ Services.obs.addObserver(function(document) {
-        if (typeof request.onsuccess === 'function') {
-          request.onsuccess.call(request, Cu.cloneInto({
-            target: {
-              result: {
-                segments: segments,
-                charsAvailableInLastSegment: charsUsedInLastSegment ?
-                  segmentLength - charsUsedInLastSegment : 0
-              }
-            }
-          }, window));
-        }
-      }, delayMs);
+      var segmentLength = 160;
+      var charsUsedInLastSegment = (text.length % segmentLength);
+
+      Services.DOMRequest.fireSuccessAsync(request, Cu.cloneInto({
+        segments: Math.ceil(text.length / segmentLength),
+        charsAvailableInLastSegment: charsUsedInLastSegment ?
+          segmentLength - charsUsedInLastSegment : 0
+      }, window));
 
       return request;
     },
 
     sendMMS: function(parameters, options) {
-      var request = Cu.waiveXrays(Cu.cloneInto({
-        onsuccess: null,
-        onerror: null
-      }, window));
+      var request = Services.DOMRequest.createRequest(window);
 
-      window.setTimeout(function() {
-        if (typeof request.onsuccess === 'function') {
-          request.onsuccess.call(request, Cu.cloneInto({
-            target: { result: null }
-          }, window));
-        }
+      parameters.type = 'mms';
 
-        parameters.type = 'mms';
+      createMessage(parameters, options).then(function(message) {
+        Services.DOMRequest.fireSuccessAsync(request, null);
 
-        callEventHandlers('sending', Cu.waiveXrays(Cu.cloneInto({
-          message: createMessage(parameters, options)
-        }, window)));
-      }, delayMs);
+        callEventHandlers(
+          'sending', Cu.cloneInto({ message: message }, window)
+        );
+      });
 
       return request;
     },
 
     send: function(recipients, content, options) {
-      var requests = Cu.waiveXrays(Cu.cloneInto(recipients.map(() => ({
-        onsuccess: null,
-        onerror: null
-      })), window));
-
-      requests.forEach((request, index) => {
-        window.setTimeout(function() {
-          if (typeof request.onsuccess === 'function') {
-            request.onsuccess.call(request, Cu.cloneInto({
-              target: { result: null }
-            }, window));
-          }
-          var parameters = {
-            type: 'sms',
-            body: content,
-            receiver: recipients[index]
-          };
-
-          var o = new window.Object();
-          o.message = createMessage(parameters, options);
-
-          callEventHandlers('sending', o);
-        }, delayMs);
+      var requests = recipients.map((receiver) => {
+        var request = Services.DOMRequest.createRequest(window);
+
+        var parameters = {
+          type: 'sms',
+          body: content,
+          receiver: receiver
+        };
+
+        createMessage(parameters, options).then(function(message) {
+          Services.DOMRequest.fireSuccessAsync(request, null);
```

Comment:
Yes, developer who will want to cover this case should be responsible for extending mock :)

This comment should probably be at the top of the file, does it work for you?